0001-interledger-architecture/0001-interledger-architecture.md

```diff
@@ -1,136 +1,168 @@
 
-### Transport Layer
+![Diagram: a prepare packet flows forward from Sender to Connector, to another Connector, who rejects it. A reject packet flows backward from the second Connector to the first Connector, then to the Sender.](../shared/graphs/packet-flow-reject.svg)
 
-Transport layer protocols are end-to-end protocols used by the senders and receivers of Interledger payments to determine the payment condition and other details. The guarantees afforded to the sender vary depending on the type of transport protocol used.
+The flow is specified in detail in [IL-RFC-27: Interledger Protocol version 4](../0027-interledger-protocol-4/0027-interledger-protocol-4.md).
 
-There is currently one transport layer protocol:
+#### Packet Size
 
-* [STREAM](../0029-stream/0029-stream.md) 
+A packet does not have to represent the full amount of a real-world payment. _Transport protocols_ built on top of the Interledger Protocol can combine many small-value packets into a single large-value payment, rejecting and retrying individual packets as necessary to achieve the desired outcome. Using small packets in this way has several benefits for the network:
 
-    In the STREAM protocol, the sender and receiver use a shared secret to generate the payment condition, authenticate the ILP packet, and encrypt the application data. Using STREAM, the sender is guaranteed that fulfillment of their transfer indicates the receiver got the payment, provided that no one aside from the sender and receiver have the secret and the sender did not submit the fulfillment. In addition, STREAM is a multiplexed ILP transport protocol that provides a full-duplex communication channel for money and data between endpoints. It also automatically determines how much money and data can be sent in each ILP packet. 
+- Small packets lock up less money at the same time. This lets the same money be more frequently, increasing its efficiency.
+- Small packets reduce the "free option problem" where senders can lock up connectors' funds, then either take the promised deal or not depending on whether exchange rates move in their favor.
+- Small packets involve small connectors (those with less funds available) in more transactions, and lower the barrier to entry for operating a connector, since moving less than the full amount of a payment can still be useful.
+- Small packets let participants settle their balances more often, so that the outstanding balance between two participants can remain low even when sending large payments.
 
-### Interledger Layer
+The Interledger Protocol does not have a specific definition of "small", nor a size limit on packets. A rule of thumb for "how small" a packet should be is that the average payment should consist of several packets and a large payment should consist of many packets. More generally, the Interledger design works best when payments are divided into the smallest pieces that are practical to send.
```

Comment:
Agreed

I would mention that the packet size is an emergent property of the network, rather than something specified by the protocol. Each connector can set their own maximum packet amount, and any path's effective maximum is the minimum of the connectors along that path.

Another benefit of building for small packets is that you can better handle the inevitability of different connectors having different packet sizes. Making them more homogeneous lowers the chance that one connector's minimum will be bigger than another's maximum amount (which would fragment the network and break interoperability)

Reply:
It doesn't sit well with me to suggest a non-quantifiable value like "small" with no guidance whatsoever as to its meaning.

I also don't think it's entirely useful to say that the packet size is an emergent property of the network. This may be true, but it emerges from the collective intentions of network users, so it only seems fair to state what sorts of concerns those users are thinking about when they choose their packet sizes.

So far, the best I've got is to change this paragraph to read as follows:

> The Interledger Protocol does not have a specific definition of "small", nor a size limit on packets. Each connector can choose minimum and maximum packet sizes they are willing to relay; as a result, any path's maximum packet size is the smallest maximum packet size among the connectors in that path. To be compatible with as much of the network as possible, one should choose packet sizes that fit between the minimum and maximum values of as many connectors as possible.

Reply:
> So far, the best I've got is ...

I think that is pretty good. Perhaps we should reference the analogy of path MTU in IP-based networks?

Also, I think we should standardise on saying "maximum packet value" or "maximum packet amount" since the packet does also have a data payload which is more accurately described by "size".

@emschwartz WDYT?

Also could add something like:

It is possible to discover the maximum packet amount of a path by sending probing packets of increasingly larger amounts until these begin to be rejected. This technique is currently employed by implementations of the STREAM protocol.